src/paths/dijkstra.c

```diff
@@ -941,36 +972,77 @@ int igraph_get_all_shortest_paths_dijkstra(const igraph_t *graph,
         /* now, reconstruct the shortest paths from the parent list in the
          * order we've found the nodes during the traversal.
          * dists is being re-used as a vector where element i tells the
-         * index in res where the shortest paths leading to vertex i
+         * index in vertices where the shortest paths leading to vertex i
          * start, plus one (so that zero means that there are no paths
          * for a given vertex).
          */
         paths_index = &dists;
         n = igraph_vector_size(&order);
         igraph_vector_null(paths_index);
 
-        /* clear the paths vector */
-        igraph_vector_ptr_clear(res);
-        res_item_destructor = igraph_vector_ptr_get_item_destructor(res);
-        igraph_vector_ptr_set_item_destructor(res,
-                                              (igraph_finally_func_t*)igraph_vector_destroy);
+        if (edges){
+            igraph_vector_ptr_clear(edges);
+            res_item_destructor = igraph_vector_ptr_get_item_destructor(edges);
```

Comment:
Note that `res_item_destructor` was used to store, and later revert, the item destructor of `res`, which you now renamed `vertices`. Thus, now we need to separately save and restore the item destructor of both `vertices` and `edges`.